src/ThirdPartyProvider.js

```diff
@@ -42,24 +41,8 @@ export class ThirdPartyProvider {
      */
     providerStatusUpdated (data) {
         try {
-            const {credentials, availableInputTypes} = validate(data, providerStatusUpdatedSchema)
-
-            // Update local settings and data
-            this.device.settings.setAvailableInputTypes(availableInputTypes)
-            this.device.storeLocalCredentials(credentials)
-
-            // rerender the tooltip
-            this.device.uiController?.updateItems(credentials)
-
-            if (!this.device.globalConfig.isTopFrame) {
-                // If the tooltip is open on an autofill type that's not available, close it
-                const currentInputSubtype = getSubtypeFromType(this.device.getCurrentInputType())
-                if (!availableInputTypes.credentials?.[currentInputSubtype]) {
-                    this.device.removeTooltip()
-                }
-                // Redecorate fields according to the new types
-                this.device.scanner.forms.forEach(form => form.recategorizeAllInputs())
-            }
+            const { availableInputTypes, credentials } = validate(data, providerStatusUpdatedSchema)
+            this.device.updateAutofillInputs(availableInputTypes, credentials)
```

Comment:
Nice!

Reply:
This is what was refactored, I can keep it too if you say nice :D

Reply:
If it's no longer needed, then let's revert. If it's not broken …